### DISCUSSION TITLE
CSS-TRICKS - If we’re gonna criticize utility-class frameworks, let’s be fair about it

### DISCUSSION THREAD

**Discussion by jenstornell** [tailwindlabs/tailwindcss#4760] [General]
An interesting article on CSS-TRICKS about Tailwind and other utility-class frameworks.

https://css-tricks.com/if-were-gonna-criticize-utility-class-frameworks-lets-be-fair-about-it/

**Comment by Martleby** [comment]
I'm always surprised to see that specific complaint about not liking the number of classes on an element. To me it makes much more sense to have composable classes on the markup it applies to and then to abstract the whole thing to a component. I also find it far easier than wading through tons of css to find the .card, work out what it's doing, then find the media queries to work out what they do etc.

Mind you I've been using Tailwind for so long now that I don't think I could ever go back to pure CSS (or a preprocessor). Tailwind has just become second nature to me now 🥳

**Reply by jenstornell** [reply]
I don't mind the number of classes but sometimes I still use the `@apply` and make a custom class for reusability. When I use what depends on the situation. Sometimes it works fine with just a component approach instead.

I also think Tailwind is getting better for each version. More complete, less hassle and more simple than ever to install.

The thing I like the most with Tailwind is that I don't need to jump back and forth between the html and the css file. I can stay in the html file all the time, even for hover states etc.

The people who don't know Tailwind probably only think it's like 10% as powerful as it really is. At least is was like that for me before getting into it.
